packages/lesswrong/components/seasonal/PetrovDayButton.tsx

```diff
@@ -157,56 +188,85 @@ const PetrovDayButton = ({classes, refetch}: {
-                    </LWTooltip>
-                    :
-                    <LWTooltip title="Are you sure?" placement="right">
-                      <div onClick={pressButton}>
-                        <img className={classes.buttonDefault} src={"../petrovButtonUnpressedDefault.png"}/>
-                        <img className={classes.buttonHover} src={"../petrovButtonUnpressedHover.png"}/>
-                      </div>
-                    </LWTooltip>
-                  }
-                </div>
-              :
+          }
+          {currentUser ? 
               <div className={classes.button}>
-                <LoginPopupButton title={"Log in if you'd like to push the button"}>
-                  <div>
-                    <img className={classes.buttonDefault} src={"../petrovButtonUnpressedDefault.png"}/>
-                    <img className={classes.buttonHover} src={"../petrovButtonUnpressedHover.png"}/>
-                  </div>
-                </LoginPopupButton>
+                {(renderButtonAsPressed || launched) ? 
+                  <LWTooltip title={<div><div>You have pressed the button.</div><div>You cannot un-press it.</div></div>} placement="right">
+                    <img className={classes.buttonPressed} src={"../petrovButtonPressedDark.png"}/> 
+                  </LWTooltip>
+                  :
+                  <LWTooltip title="Are you sure?" placement="right">
+                    <div onClick={pressButton}>
+                      <img className={classes.buttonDefault} src={"../petrovButtonUnpressedDefault.png"}/>
+                      <img className={classes.buttonHover} src={"../petrovButtonUnpressedHover.png"}/>
+                    </div>
+                  </LWTooltip>
+                }
+              </div>
+            :
+            <div className={classes.button}>
+              <LoginPopupButton title={"Log in if you'd like to push the button"}>
+                <div>
+                  <img className={classes.buttonDefault} src={"../petrovButtonUnpressedDefault.png"}/>
+                  <img className={classes.buttonHover} src={"../petrovButtonUnpressedHover.png"}/>
+                </div>
+              </LoginPopupButton>
+            </div>
+          }
+  
+          {launched ?
+            <div className={classes.info}>
+              <p >{isEAForum ? "The EA Forum's" : "LessWrong's"} missiles have been launched.</p>
+              <p>We hope you made good choices.</p>
+            </div>
+            : <div className={classes.inputSection}>
+              {renderButtonAsPressed && <TextField
+                onChange={updateLaunchCode}
+                placeholder={"Enter Launch Codes"}
+                margin="normal"
+                variant="outlined"
+              />}
+              {(renderLaunchButton) && 
+                <Button onClick={launch} className={classes.launchButton} disabled={!!(currentUser as any).petrovCodesEntered}>
```

Comment:
Can you view this while logged out?

Reply:
Why'd you have to type assert here? It seems like it should have worked. Also our convention is that you always put a comment when doing type assertions.